### PR TITLE
Improve aggregation and extraction tests

### PR DESCRIPTION
**Pull request by adelowo**
Extra test suites for https://github.com/blndgs/model/pull/52 trying to catch as much edge cases

## Summary by CodeRabbit

- **New Features**
	- Enhanced signature validation process for improved efficiency.
	- Added multiple new test functions to improve error handling and validation for user operations.

- **Bug Fixes**
	- Improved error handling in user operation tests, covering various scenarios including empty and invalid data.

### CODE REVIEW

**Comment by blewater** [comment]
@coderabbitai review

File: userops_ext.go [hunk 1]
```diff
@@ -411,7 +411,10 @@ func (op *UserOperation) SetIntent(intentJSON string) error {
 func (op *UserOperation) GetSignatureValue() []byte {
 	if no0xPrefix(op.Signature) {
 		lenSig := len(op.Signature)
-		if lenSig >= KernelSignatureLength && sigHasKernelPrefix(op.Signature) {
+
+		// sigHasKernelPrefix already checks this internally
+		// if lenSig >= KernelSignatureLength && sigHasKernelPrefix(op.Signature) {
+		if sigHasKernelPrefix(op.Signature) {
```
Comment (blewater) [inline userops_ext.go:417]: 👍

File: userops_ext_test.go [hunk 2]
```diff
@@ -1892,3 +1893,1269 @@ func TestDataCopy_SetCrossChainIntent(t *testing.T) {
...
+	require.NoError(t, err)
+	require.Equal(t, UnsolvedAggregateUserOp, status)
+
+	// Extract and verify the new embedded operation matches embedOp2
+	extractedOp, err := baseOp.ExtractEmbeddedOp()
+	require.NoError(t, err)
+	require.Equal(t, embedOp2.Nonce.String(), extractedOp.Nonce.String())
+}
+
+func TestUserOperation_Aggregate_ErrorHandling(t *testing.T) {
```
Comment (blewater) [inline userops_ext_test.go:2998]: A test aggregating an aggregate userOp? 
Any result is acceptable. A clear error message or double aggregation/extraction.

File: userops_ext_test.go [hunk 3]
```diff
@@ -1892,3 +1893,1269 @@ func TestDataCopy_SetCrossChainIntent(t *testing.T) {
 	// require.NotEqual(t, originalCallData[0], op.CallData[0])
 	require.Equal(t, originalCallData[1:], op.CallData[1:])
 }
+
```
Comment (blewater) [inline userops_ext_test.go:1896]: A few test suggestions:

- Add a test(s) attempting to solve (set EVM solution) an aggregate object. An error should be returned.
- Simulate positive solving process: Create an aggregate object, extract the packed op, and solve the op.
- Any tests that attempt to extract op with agg op that has incorrect packedOpsLength (0 or > 1).

**Review by blewater — CHANGES_REQUESTED** [review]
Great! I have a few more suggestions to test the error messages.

**Review by blewater — APPROVED** [review]
LGTM!
